File: clustering-service/src/models/cluster.py
from dataclasses import dataclass
import datetime
from typing import Optional

@dataclass
class Cluster:
    id: str
    contractAddress: str
    createdAt: datetime.date
    updatedAt: datetime.date
    tokenIds: list[str]
    rankAverage: Optional[float] = None
    totalVolume: Optional[float] = None
    totalSalesByMarketplace: Optional[dict[str, float]] = None
    latestSaleBlockNumber: Optional[int] = None
    attributes: Optional[dict[str, dict[str, int]]] = None

File: gql-service/src/models/cluster.ts
export interface Cluster {
  id: string;
  contractAddress: string;
  createdAt?: string;
  updatedAt?: string;
  tokenIds: string[];
  rankAverage?: number;
  totalVolume?: number;
  totalSalesByMarketplace: { [key: string]: number };
  latestSaleBlockNumber: number;
  attributes: { [key: string]: { [key: string]: number } };
}
